core/source/general/block_array.hpp

```diff
@@ -13,6 +13,7 @@
 #define BLOCK_ARRAY_HPP
 
 #include "general/debug.h"
+#include <cstring>
```

Comment:
I don't think cstring should be included here; maybe it needs to be in another field that includes block_array.hpp?